client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/evaluation/classification/AucRocMetric.java

```diff
@@ -0,0 +1,267 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.ml.dataframe.evaluation.classification;
+
+import org.elasticsearch.client.ml.dataframe.evaluation.EvaluationMetric;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Area under the curve (AUC) of the receiver operating characteristic (ROC).
+ * The ROC curve is a plot of the TPR (true positive rate) against
+ * the FPR (false positive rate) over a varying threshold.
+ */
+public class AucRocMetric implements EvaluationMetric {
+
+    public static final String NAME = "auc_roc";
+
+    public static final ParseField INCLUDE_CURVE = new ParseField("include_curve");
+    public static final ParseField CLASS_NAME = new ParseField("class_name");
+
+    @SuppressWarnings("unchecked")
+    public static final ConstructingObjectParser<AucRocMetric, Void> PARSER =
+        new ConstructingObjectParser<>(NAME, true, args -> new AucRocMetric((Boolean) args[0], (String) args[1]));
+
+    static {
+        PARSER.declareBoolean(optionalConstructorArg(), INCLUDE_CURVE);
+        PARSER.declareString(optionalConstructorArg(), CLASS_NAME);
+    }
+
+    public static AucRocMetric fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    public static AucRocMetric withCurveForClass(String className) {
+        return new AucRocMetric(true, className);
+    }
+
+    public static AucRocMetric forClass(String className) {
+        return new AucRocMetric(false, className);
+    }
+
+    private final boolean includeCurve;
+    private final String className;
+
+    public AucRocMetric(Boolean includeCurve, String className) {
+        this.includeCurve = includeCurve == null ? false : includeCurve;
+        this.className = className;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder
+            .startObject()
+            .field(INCLUDE_CURVE.getPreferredName(), includeCurve);
+        if (className != null) {
```

Comment:
Actually this field is required for the metric to make sense.
Changed `class_name` to non-optional and changed the order of parameters so that the now-required `class_name` comes first (before optional `include_curve`).